Apps/phradar/radar_connector.py

```diff
@@ -0,0 +1,355 @@
+
+        # use timezone set in asset configuration to set discovered date_time timezone
+        config = self.get_config()
+        incident_group = int(config.get("radar_incident_group"))
+        time_zone = config.get("time_zone", "UTC")
+        date_time = datetime.now(timezone(time_zone)).strftime("%Y-%m-%dT%H:%M:%S")
+
+        # get incident channel information
+        phantom_base_url = self._get_system_settings()["company_info_settings"]["fqdn"]
+        container_path = ""
+        container_id = self.get_container_id()
+        if container_id is not None:
+            container_path = f"/mission/{container_id}"
+
+        uri = f"{phantom_base_url}{container_path}"
+
+        self.save_progress(f"create payload")
+
+        body = {
+            "incident_group_id": incident_group,
+            "channel": {
+                "id": f"{container_id}",
+                "source": "Splunk Phantom",
+                "uri": uri
+            },
+            "name": name,
+            "description": description,
+            "discovered": {
+                "date_time": date_time,
+                "time_zone": time_zone
+            }
+        }
+
+        radar_val, data = self._make_rest_call("/incidents", action_result, method="post", data=json.dumps(body))
+
+        if phantom.is_fail(radar_val):
+            return action_result.get_status()
+
+        data["incident_id"] = data["id"]
+        data["name"] = name
+        data["description"] = description
+        data["group"] = incident_group
+        data["discovered"] = f"{date_time} / {time_zone}"
+
+        self.save_progress("add data to action result")
+        action_result.add_data(data)
+
+        # Return success, no need to set the message, only the status
+        return action_result.set_status(phantom.APP_SUCCESS)
+
+    def _get_system_settings(self):
+        self.save_progress("get system settings")
+
+        url = f"{self.get_phantom_base_url()}rest/system_settings"
+        try:
+            resp = requests.get(url, verify=False)
+            resp_json = resp.json()
+        except Exception as e:
+            self.debug_print("Unable to query system settings: ", e)
+
+        return resp_json
+
+    def _handle_add_note_to_privacy_incident(self, param):
+        self.save_progress(f"run action {self.get_action_identifier()}")
+
+        # Add an action result object to self (BaseConnector) to represent the action for this param
+        action_result = self.add_action_result(ActionResult(dict(param)))
+        content = param.get("content")
+        category = "Splunk Phantom"
+        incident_id = param.get("incident_id")
+
+        body = {
+            "content": content,
+            "category": category,
+        }
+
+        # make rest call
+        radar_val, data = self._make_rest_call(f"/incidents/{param.get('incident_id')}/notes", action_result, method="post", data=json.dumps(body))
```

Comment:
recommend you add a debug_print to dump the parameters here

Reply:
Just to clarify on this point, I did add `debug_print` statements for exception handling in response to this recommendation, but not add one here since it added a lot more noise to the logs.